adapters/facebook_test.go

```diff
@@ -106,10 +108,26 @@ func DummyFacebookServer(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, fmt.Sprintf("No banner object sent"), http.StatusInternalServerError)
 			return
 		}
-		if breq.Imp[0].Banner.W != 300 || breq.Imp[0].Banner.H != 250 {
-			http.Error(w, fmt.Sprintf("Size '%dx%d' not supported", breq.Imp[0].Banner.W, breq.Imp[0].Banner.H), http.StatusInternalServerError)
+		if breq.Imp[0].Instl == 0 {
+			supportedHeight := map[uint64]bool{
+				50:  true,
+				90:  true,
+				250: true,
+			}
+			if !supportedHeight[breq.Imp[0].Banner.H] {
+				http.Error(w, fmt.Sprintf("Height '%d' not supported", breq.Imp[0].Banner.H), http.StatusInternalServerError)
```

Comment:
Are these actually the status codes that the facebook server sends back, if you give them unsupported values?

I would've expected a 4xx of some sort... if their server really does respond with a 5xx, though, this is of course fine.

Reply:
If we send unsupported sizes, facebook would return bids that won't be rendered in the SDK, so not necessarily a 5xx response from them. What we want is once we found invalid sizes, stop sending the request to facebook and return an error invalid request. Not sure how to raise a 4xx error.

Reply:
It would be the third argument. `http.Error(..., ..., http.StatusBadRequest)`.

To be clear, though... I _don't_ think prebid-server should be responsible for filtering bids because the client can't handle it. We should accept any bid params that a particular bidder allows, and return all the bids which their server returns. If a client can't handle it, it's their responsibility to filter the bids before rendering.

Otherwise we'll be in a situation where (for example) Mobile/Apps can't handle these bids, but JS/Browsers can. If the server filters bids because one client can't handle it, then the other client gets limited too.

So... the mock server in these tests should respond the same way that facebook's _real_ server does, when you send it requests like these. If they respond with a 5xx, this should use `http.StatusInternalServerError`. If they respond with a 4xx, this should use`http.StatusBadRequest`. If they respond with actual bids or an empty array, this should use `http.StatusOK` with some similarly structured data.

I've never run a request against their actual server, though, so... not sure which one is actually right here. I just mention it because a `5xx` would be a very unusual choice on their part.